timemachine/cpp/src/random_sampler.cu

```diff
@@ -0,0 +1,89 @@
+#include "device_buffer.hpp"
+#include "gpu_utils.cuh"
+#include "kernels/k_sampling.cuh"
+#include "kernels/kernel_utils.cuh"
+#include "math_utils.cuh"
+#include "random_sampler.hpp"
+
+namespace timemachine {
+
+template <typename RealType>
+RandomSampler<RealType>::RandomSampler(const int N, const int seed)
+    : N_(N), temp_storage_bytes_(0), d_rand_(round_up_even(N_)), d_gumbel_(N_) {
+
+    cudaSafeMalloc(&d_arg_max_, 1 * sizeof(cub::KeyValuePair<int, RealType>));
+    gpuErrchk(cub::DeviceReduce::ArgMax(
+        nullptr, temp_storage_bytes_, d_gumbel_.data, static_cast<cub::KeyValuePair<int, RealType> *>(d_arg_max_), N_));
```

Comment:
`static_cast` or `reinterpret_cast`? Not sure why you're allowed to statically cast a `void *` into a `cub::KVPair*`

Reply:
Removed after @proteneer  suggested trying to import `#include <cub/util_type.cuh>` which works: https://github.com/proteneer/timemachine/pull/1195/commits/12d5302dbc42a52a23d2904909af4605e9da2f16